packages/@react-stately/calendar/src/useCalendarState.ts

```diff
@@ -196,7 +196,7 @@ export function useCalendarState<T extends DateValue>(props: CalendarStateOption
       return isFocused && focusedDate && isSameDay(date, focusedDate);
     },
     isCellDisabled(date) {
-      return props.isDisabled || date.compare(startDate) < 0 || date.compare(endDate) > 0 || isInvalid(date, minValue, maxValue);
+      return props.isDisabled || date.compare(startDate) < 0 || date.compare(endDate) > 0 || isInvalid(date, minValue, maxValue) || props.isDateDisabled && props.isDateDisabled(date);
```

Comment:
Mind adding parens for clarity?
```suggestion
      return props.isDisabled || date.compare(startDate) < 0 || date.compare(endDate) > 0 || isInvalid(date, minValue, maxValue) || (props.isDateDisabled && props.isDateDisabled(date));
```